test/test_rb.py

```diff
@@ -0,0 +1,157 @@
+            qc.remove_final_measurements()
+            # Checking if the matrix representation is the identity matrix
+            self.assertTrue(
+                matrix_equal(Clifford(qc).to_matrix(), np.identity(2 ** num_qubits)),
+                "Clifford sequence doesn't result in the identity matrix.",
+            )
+
+    def validate_metadata(self, circuits: list, exp_attributes: dict):
+        """
+        Validate the fields in "metadata" for the experiment.
+        Args:
+            circuits (list): A list containing quantum circuits
+            exp_attributes (dict): A dictionary with the experiment variable and values
+        """
+        for ind, qc in enumerate(circuits):
+            self.assertTrue(
+                qc.metadata["xval"] == exp_attributes["lengths"][ind],
+                "The number of gates in the experiment metadata doesn't match to the one provided.",
+            )
+            self.assertTrue(
+                qc.metadata["qubits"] == tuple(exp_attributes["qubits"]),
+                "The qubits indices in the experiment metadata doesn't match to the one provided.",
+            )
+
+    def validate_circuit_data(
+        self,
+        experiment: qe.randomized_benchmarking.rb_experiment.RBExperiment,
+        exp_attributes: dict,
+    ):
+        """
+        Validate that the metadata of the experiment after it had run matches the one provided.
+        Args:
+            experiment: The experiment data and results after it run
+            exp_attributes (dict): A dictionary with the experiment variable and values
+        """
+        self.assertTrue(
+            exp_attributes["lengths"] == experiment.experiment_options.lengths,
+            "The number of gates in the experiment doesn't match to the one in the metadata.",
+        )
+        self.assertTrue(
+            exp_attributes["num_samples"] == experiment.experiment_options.num_samples,
+            "The number of samples in the experiment doesn't match to the one in the metadata.",
+        )
+        self.assertTrue(
+            tuple(exp_attributes["qubits"]) == experiment.physical_qubits,
+            "The qubits indices in the experiment doesn't match to the one in the metadata.",
+        )
+
+    def _exp_data_properties(self):
+        """
+        Creates a list of dictionaries that contains invalid experiment properties to check errors.
+        The dict invalid data is as following:
+            exp_data_list[1]: same index of qubit.
+            exp_data_list[2]: qubit index is negative.
+            exp_data_list[3]: the length of the sequence has negative number.
+            exp_data_list[4]: num of samples is negative.
+            exp_data_list[5]: num of samples is 0.
+            exp_data_list[6]: the length of the sequence list has duplicates.
+        Returns:
+            list[dict]: list of dictionaries with experiment properties.
+        """
+        exp_data_list = [
```

Comment:
I think that the most common mistakes are when `"lengths"` is not strictly increasing, and in particular when the same length appears twice. This can happen if someone generates the length sequence using some function (e.g. a logarithmic function), see for example the discussion here:
https://github.com/Qiskit/qiskit-ignis/issues/459
I never saw errors when the `"qubits"` or `"length"` are non-positive (or non-integers)